src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -76,32 +76,50 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
 
-                        // Need to use FQDN normalized host so that CNAME's are traversed.
-                        // Use DNS to do the forward lookup to an A (host) record.
-                        // But skip DNS lookup on IP literals. Otherwise, we would end up
-                        // doing an unintended reverse DNS lookup.
-                        string spn;
-                        UriHostNameType hnt = authUri.HostNameType;
-                        if (hnt == UriHostNameType.IPv6 || hnt == UriHostNameType.IPv4)
+                        // Calculate SPN (Service Principal Name) using the host name of the request.
+                        // Use the request's 'Host' header if available. Otherwise, use the request uri.
+                        string hostName;
+                        if (request.HasHeaders && request.Headers.Host != null)
                         {
-                            spn = authUri.IdnHost;
+                            // Use the host name without any normalization.
+                            hostName = request.Headers.Host;
+                            if (NetEventSource.IsEnabled)
+                            {
+                                NetEventSource.Info(connection, $"Authentication: {challenge.AuthenticationType}, Host: {hostName}");
+                            }
                         }
                         else
                         {
-                            IPHostEntry result = await Dns.GetHostEntryAsync(authUri.IdnHost).ConfigureAwait(false);
-                            spn = result.HostName;
+                            // Need to use FQDN normalized host so that CNAME's are traversed.
+                            // Use DNS to do the forward lookup to an A (host) record.
+                            // But skip DNS lookup on IP literals. Otherwise, we would end up
+                            // doing an unintended reverse DNS lookup.
+                            UriHostNameType hnt = authUri.HostNameType;
+                            if (hnt == UriHostNameType.IPv6 || hnt == UriHostNameType.IPv4)
+                            {
+                                hostName = authUri.IdnHost;
+                            }
+                            else
+                            {
+                                IPHostEntry result = await Dns.GetHostEntryAsync(authUri.IdnHost).ConfigureAwait(false);
+                                hostName = result.HostName;
+                            }
                         }
-                        spn = "HTTP/" + spn;
 
+                        string spn = "HTTP/" + hostName;
                         if (NetEventSource.IsEnabled)
                         {
-                            NetEventSource.Info(connection, $"Authentication: {challenge.AuthenticationType}, Host: {authUri.IdnHost}, SPN: {spn}");
+                            NetEventSource.Info(connection, $"Authentication: {challenge.AuthenticationType}, SPN: {spn}");
```

Comment:
Could we just have one NetEventSource.Info call that logs uri, hostname, and spn, rather than three separate ones?

Reply:
I had separate calls because I was trying to capture the logic of how the SPN was assembled (i.e. host header present or not, DNS resolution, etc.).  But you raise a good point that we now have too many separate log entries for this.

I'll submit a follow-up PR to try to address this.